content/post/practical_ml_2.md

```diff
@@ -0,0 +1,55 @@
+ 
+However, consider our spam filter. Every time a user sends an event we must classify it as spam or not. Our filter is part of a series of checkpoints between when the user hits *send* and when the email gets delivered in their recipient's inbox. That's very real-time. In this paradigm, optimization and efficiency are critical. If we trained a heavy model, however accurate, that slowed down our email deliverability, it would never see the light of day. Moreover, since this classifier interacts with the application, it must in some way be integrated with the greater application infrastructure. This means more scrutiny to testing, exception handling, logging and other software engineering best practices. 
+ 
+In this context, the Data Scientist quickly realizes that her statistical model is but a very small part in a large web of interconnected processes. As a result, it may be prudent to spend less time tweaking a model to achieve some theoretical guarantee and more time understanding how it will fit in the bigger picture.
+ 
+## Show and tell
+ 
+I am a strong proponent of frequent presentations, discussions and knowledge sharing. This is especially important when working on a large, complex Machine Learning model. Presenting one's work has the following benefits:
+ 
+* *Avoid rabbit holes*: Periodically showing your progress to Stakeholders or peers forces you out of the weeds and helps you to summarize results. Often, this self-organization and peer feedback will help you to realize that a particular rabbit hole may not be worth the effort. 
+* *Reinforce understanding*: There is no better indication that you have a solid understanding of your work than being able to clearly explain it to someone else, especially those less technical.
+* *Forced articulation*: Model building involves making numerous assumptions along the way. Sometimes simplifying assumptions are made in order not to lose momentum during the model building process. It is critical to return to those assumptions and correct them if necessary, lest they grow into trouble areas if your model becomes very sensitive to them. Presentations form a natural stopping point, forcing you to address your assumptions. It it foolhardy to present work with half-baked assumptions that you didn't account for ahead of time.
+* *New perspectives*: The Data Scientist should be the expert in building a good Machine Learning model. But even experts can [learn from others](https://www.locallyoptimistic.com/post/practical_ml/). Do not underestimate the value in getting a different perspective.
+* *Show off*: Business leaders are not generally enamored of siloed functions, especially functions that are esoteric and whose value is difficult to quantify. Data Science functions can sometimes fall into this category partly because in general we are not great at selling ourselves, explaining our methodologies or communicating how we add value. Presentations are a great way to counter that.
```

Comment:
Is it "enamored with" or "enamored by"? I am not an expert here, but these two phrases sound more natural to me.

Reply:
https://english.stackexchange.com/questions/48891/enamored-of-with-by

I wasn't sure either. It seems that "with" or "of" is more popular. Happy to make it "with".